avae/data.py

```diff
@@ -151,39 +126,45 @@ class ProteinDataset(Dataset):
     """
 
     def __init__(
-        self, root_dir, amatrix=None, transform=None, lim=None, collect_m=False
+        self,
+        root_dir,
+        amatrix=None,
+        classes=None,
+        transform=None,
+        lim=None,
+        collect_m=False,
     ):
         super().__init__()
 
         self.collect_meta = collect_m
 
+        self.amatrix = amatrix
+
         self.root_dir = root_dir
         self.paths = [f for f in os.listdir(root_dir) if ".mrc" in f]
         random.shuffle(self.paths)
+        ids = np.unique([f.split("_")[0] for f in self.paths])
 
-        if "classes.csv" in os.listdir(self.root_dir):
-            with open(
-                os.path.join(self.root_dir, "classes.csv"), newline="\n"
-            ) as f:
-                class_list = np.asarray(f.readlines()).flatten()
-                class_list = [
-                    c.strip() for c in class_list if len(c.strip()) != 0
-                ]
+        if self.amatrix is not None:
+            class_check = np.in1d(ids, self.amatrix.columns)
+            if not np.all(class_check):
+                raise RuntimeError(
+                    "Not all classes in the training set are present in the "
+                    "affinity matrix. Missing classes: {}".format(
+                        np.asarray(ids)[~class_check]
+                    )
+                )
 
-            self.paths = [p for p in self.paths for c in class_list if c in p]
+        if classes is not None:
+            with open(classes, newline="\n") as f:
+                classes = np.asarray(f.read().splitlines()).flatten()
+                classes = [c for c in classes if c != ""]
+                classes = [c.strip() for c in classes if len(c.strip()) != 0]
```

Comment:
In a chat with @marjanfamili yesterday we thought it could be useful to have the list of final classes as an attribute to the ProteinDataset, so it can be accessed later (in [here](https://github.com/alan-turing-institute/affinity-vae/blob/705f4850990165f5bb7faa6bc4db75be1d4db89d/avae/data.py#L40) to visualise the matrix for the given classes). It could be defined after this block of code as `self.final_classes = classes` if there is a classes.csv file, or `self.final_classes = ids`. (doesn't have to be called final_classes btw, is just to avoid confusion with the other variables).

Reply:
That sounds good, adding self.final_classes attribute.